include/ringmesh/mesh/mesh_aabb.h

```diff
@@ -115,6 +134,23 @@ namespace RINGMesh
         std::tuple< index_t, vecn< DIMENSION >, double > closest_triangle(
             const vecn< DIMENSION >& query ) const;
 
+        /*!
+         * @brief Gets the closest triangle to a given point using the
+         * given distance \p EvalDistance
+         * @pre The mesh needs to be triangulated
+         * @param[in] query the point to use
+         * @return a tuple containing:
+         * - the closest triangle index.
+         * - the nearest point on the closest triangle.
+         * - the distance between \p query and \p nearest_point.
+         */
+        template < typename EvalDistance >
```

Comment:
Can you add a test checking that this distance when used to get the L2 norm gives the same result as the L2 norm function?
Can you add a test to test some norms?

Reply:
Good idea. I will.

Reply:
Done